File: iossdkdemo/SDKDemo/APIControllers/DemoSurveyViewController.m
//
//  DemoSurveyViewController.m
//  SDKDemoApp
//
//  Created by OnePoint Global on 06/10/16.
//  Copyright © 2016 opg. All rights reserved.
//

#import "DemoSurveyViewController.h"
#import <OPGSDK/OPGSDK.h>
#import "TakeSurveyViewController.h"
#import "SurveyListViewController.h"

@interface DemoSurveyViewController ()
@property(strong, nonatomic) NSArray* surveyList;
@property(strong, nonatomic) NSArray* panelList;
@end

@implementation DemoSurveyViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self.activityIndicator setHidden:TRUE];
    self.txtSharedKey.delegate = self;
    self.txtUsername.delegate = self;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


#pragma mark - IBAction
-(IBAction)btnGOTapped:(id)sender{
    
    if ((self.txtUsername.text == nil || [self.txtUsername.text isEqualToString:@""]) || (self.txtSharedKey.text == nil || [self.txtSharedKey.text isEqualToString:@""])) {
        // pick the default admin name and key from Appdelegate
    }
    else{
        // Initialising OPGSDK with admin name and key from Text fileds
        //[OPGSDK initializeWithUserName:self.txtUsername.text withSDKKey:self.txtSharedKey.text];
        
    }
    OPGSDK* sdk = [OPGSDK new];
    NSError *error;
    OPGAuthenticate *obj=[sdk authenticate:self.txtUsername.text password:self.txtSharedKey.text error:&error];
    self.surveyList = [sdk getUserSurveyList:&error];       // Getting list of Surveys assigned to user
    if (error == nil) {
        [self performSegueWithIdentifier:@"embedSurveyList" sender:sender];
    }
    else{
        NSLog(@"Error - %@",[error description]);
        UIAlertController * alert=   [UIAlertController
                                      alertControllerWithTitle:@"MySurveys Demo"
                                      message:[[error userInfo] valueForKey:NSLocalizedDescriptionKey]          /* @"Error Occured. Contact Support!" */
                                      preferredStyle:UIAlertControllerStyleAlert];
        
        UIAlertAction* noButton = [UIAlertAction
                                   actionWithTitle:@"Got it!"
                                   style:UIAlertActionStyleDefault
                                   handler:^(UIAlertAction * action)
                                   {
                                       //Handle error
                                       
                                   }];
        [alert addAction:noButton];
        
        [self presentViewController:alert animated:YES completion:nil];
        
    }
}


#pragma mark - Segues Operations
-(void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    if ([[segue identifier] isEqualToString:@"embedSurveyList"]) {
        // Get the SurveyList view
        SurveyListViewController *viewController = [segue destinationViewController];
        viewController.surveyListArray = self.surveyList;
    }
    
}


#pragma mark - TextField delegates
-(BOOL) textFieldShouldReturn:(UITextField *)textField{
    [textField resignFirstResponder];
    return YES;
}

@end

File: iossdkdemo/SDKDemo-Swift/APIControllers-Swift/DemoSurveyViewController.swift
//
//  DemoSurveyViewController.swift
//  SDKDemoApp
//
//  Created by OnePoint Global on 06/10/16.
//  Copyright © 2016 opg. All rights reserved.
//

import UIKit

class DemoSurveyViewController: UIViewController, UITextFieldDelegate
{

    @IBOutlet weak var txtUsername : UITextField!
    @IBOutlet weak var txtSharedKey : UITextField!
    var surveyList : NSArray = []
    var obj:OPGAuthenticate!
    override func viewDidLoad() {
        super.viewDidLoad()
        self.txtSharedKey.delegate=self
        self.txtUsername.delegate=self
        // Do any additional setup after loading the view, typically from a nib.
        //self.activityIndicator.hidesWhenStopped = true
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func textFieldShouldReturn(_ textField: UITextField) -> Bool {   //delegate method
        textField.resignFirstResponder()
        return true
    }
    
    @IBAction func btnGOTapped(_ sender:AnyObject)
    {
        let sdk = OPGSDK()        // Creating OPGSDK instance
        if ((self.txtUsername.text?.characters.count)!>0 && (self.txtSharedKey.text?.characters.count)!>0)
        {
            do {
                obj = try sdk.authenticate(self.txtUsername.text, password: self.txtSharedKey.text) as OPGAuthenticate
            }
            catch{
                print("Authentication not working")         /* @"Error Occured. Contact Support!" */
                
            }
            
            do {
                self.surveyList = try sdk.getUserSurveyList() as NSArray        // Getting list of Surveys assigned to user
            }
            catch{
                print("User Survey not working")         /* @"Error Occured. Contact Support!" */
                
            }
            self.performSegue(withIdentifier: "embedSurveyList", sender: sender)
        }
        else
        {
            print("Username and Password can not be empty")
        }
        
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if(segue.identifier == "embedSurveyList")
        {
            let surveyListVC : SurveyListViewController  = segue.destination as! SurveyListViewController
            surveyListVC.surveyListArray = self.surveyList
        }
        
    }


}
